### PR TITLE
Webclient: Handle firing an event once

### PR DESCRIPTION
**Pull request by brentmclark**
## Related Ticket(s)
- [Double-clicking Login triggers multiple connect attempts](https://trello.com/c/zhJfJiSy/14)

## Short roundup of the initial problem

Double-clicking Login triggers multiple connect attempts

## What will change with this Pull Request?
- installed and configured `react-testing-library`
- Added custom hook for handle single-fire events
- Added tests for custom hook

## Screenshots
It's difficult to see a button state transition in a screenshot, so [here's a video](https://share.getcloudapp.com/geup6RwK) instead.

## Additional Notes
`React Testing Library` needed to be added manually; it is not installed with the latest version of CRA despite the library saying it is.  The React documentation states that RTL needs to be installed manually, so somebody has it wrong 🙃

This PR is in draft status.  My goal is to get consensus on the approach before I invest additional time into the remainder of the implementation.

The lockfile version changed as of this PR since I'm running NPM 7; it's backwards compatible with NPM 6.

Remaining tasks are:
- [x] Wire in the `useFireOnce` hook to the `SayMessage` flow
- [x] Allow the button disabled state to be reset in response to a state change
- [x] Test coverage for the additional functionality

### CODE REVIEW

**Comment by brentmclark** [comment]
I see the tests I wrote are failing now.  Not sure what's up; I'll have a look. 🤷

File: webclient/src/containers/Account/AddToBuddies.tsx [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-import { FormKey } from 'types';
 
-const AddToBuddies = ({ handleSubmit }) => (
-  <Form onSubmit={handleSubmit}>
-    <InputAction action="Add" label="Add to Buddies" name="userName" />
+const AddToBuddies = ({ onSubmit }) => (
+  <Form onSubmit={values => onSubmit(values)}>
+    {({ handleSubmit }) => (
+      <form onSubmit={handleSubmit}>
+        <InputAction action="Add" label="Add to Buddies" name="userName" validate={() => true} />
```
Comment (seavor) [inline webclient/src/containers/Account/AddToBuddies.tsx:10]: Should this be removed?
```suggestion
        <InputAction action="Add" label="Add to Buddies" name="userName" />
```

File: webclient/src/containers/Account/AddToIgnore.tsx [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-import { FormKey } from 'types';
 
-const AddToIgnore = ({ handleSubmit }) => (
-  <Form onSubmit={handleSubmit}>
-    <InputAction action="Add" label="Add to Ignore" name="userName" />
+const AddToIgnore = ({ onSubmit }) => (
+  <Form onSubmit={values => onSubmit(values)}>
+    {({ handleSubmit }) => (
+      <form onSubmit={handleSubmit}>
+        <InputAction action="Add" label="Add to Ignore" name="userName" validate={() => true} />
```
Comment (seavor) [inline webclient/src/containers/Account/AddToIgnore.tsx:10]: Same question.
```suggestion
        <InputAction action="Add" label="Add to Ignore" name="userName" />
```

**Comment by brentmclark** [comment]
Both of those `validate` props have been removed.

**Comment by brentmclark** [comment]
This PR is all settled from my perspective.  Is there anything you need from me in order to merge it?

**Pull request merged**